src/tools/PermissionCheckerInterface.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace hiqdev\billing\hiapi\tools;
+
+use hiqdev\php\billing\customer\CustomerInterface;
+
+interface PermissionCheckerInterface
+{
+    public function checkAccess($clientId, string $permission): bool;
+
+    public function ensureCustomerCan(CustomerInterface $customer, $permission): void;
```

Comment:
```suggestion
    public function ensureCustomerCan(CustomerInterface $customer, string $permission): void;
```

Или ты тут умышленно не добавлял тип?

+ нужно описать, какой Exception ожидать если не получится